code/modules/projectiles/guns/energy.dm

```diff
@@ -43,10 +43,30 @@
 	if(.)
 		update_icon()
 
-/obj/item/gun/energy/emp_act(severity)
+/obj/item/gun/energy/emp_act(var/severity)
 	..()
+	disable_cell_temp(severity)
 	queue_icon_update()
 
+/obj/item/gun/energy/proc/disable_cell_temp(var/severity)
+	set waitfor = FALSE
+	if(!power_supply)
+		return
+	var/mob/M
+	if(ismob(loc))
+		M = loc
+	var/initial_charge = power_supply.charge
+	power_supply.charge = 0
+	if(M && loc == M)
+		to_chat(M, SPAN_DANGER("[src] locks up!"))
+		playsound(M, 'sound/weapons/smg_empty_alarm.ogg', 30)
+	sleep(severity * 20)
+	power_supply.give(initial_charge)
+	update_maptext()
+	update_icon()
+	if(M && loc == M)
```

Comment:
Same.

Reply:
This one needs to stay. The gun might be dropped while the sleep is running.